modules/apps/site/site-cms-site-initializer/src/main/resources/META-INF/resources/js/main_view/spaces/SpaceConnectedSitesModal.tsx

```diff
@@ -152,25 +138,29 @@ const SitesSelector = ({
 						{Liferay.Language.get('site')}
 					</label>
 
-					<Autocomplete
-						allowsCustomValue
-						id="siteSelector"
-						menuTrigger="focus"
-						onChange={setValue}
-						placeholder={Liferay.Language.get('select-a-site')}
-						value={value}
+					<ItemSelector<Site>
+						apiURL={`${location.origin}/o/headless-site/v1.0/sites?active=true`}
+						items={site ? [site] : []}
+						onItemsChange={(items: Site[]) => {
+							if (items.length) {
+								setSite(items[0]);
+							}
+							else {
+								setSite(null);
+							}
+						}}
 					>
-						{sites.map((site) => (
-							<Autocomplete.Item
-								key={site.id}
-								onClick={() => {
-									setSiteSelected(site);
-								}}
+						{(item: Site) => (
+							<ItemSelector.Item
+								key={item.id}
+								textValue={item.name}
 							>
-								{site.name}
-							</Autocomplete.Item>
-						))}
-					</Autocomplete>
+								{item.name_i18n[
```

Comment:
Hey @jaimelr10 I think we should still keep this as `site.name` as long as `site.name` works. The goal of this PR is just to use the ItemSelector but we don't want to change anything else unless we have to.